### PR TITLE
Feature/avgpersonalstat

### PR DESCRIPTION
**Pull request by ibixina**
- added a new feature to calculate the avg per day stat on the personalstats page
- added the feature to the settings, and added it to manifest.json

image

what it looks like ^

### CODE REVIEW

**Comment by Sashank999** [comment]
Is there any reason why this is necessary in the first place ? The average calculation for any stats is not something I have come across, even in achievement threads.

**Comment by ibixina** [comment]
i think its a really helpful feature to have. the average gives a better/simpler way of comparison between two players' stat, expecially if the difference is really big. plenty of people would find it helpful.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Contribution
+bal
```
Comment (Sashank999) [inline CONTRIBUTING.md:2]: There is no need to add anything here.

File: extension/pages/settings/settings.html [hunk 2]
```diff
@@ -1303,6 +1303,11 @@ <h2>
 					</section>
 					<section name="profile">
 						<div class="header">Profile</div>
+						<div class="option">
+							<input id="profile-avgpersonalstats" type="checkbox" />
+							<label for="profile-avgpersonalstats">Show avg Personal Stats</label>
```
Comment (Sashank999) [inline extension/pages/settings/settings.html:1308]: Please avoid using abbreviations in settings names.

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+			if (body.includes("getGraphData")) {
+				const graphData = json;
+				calculateStatsAverage(graphData);
+			}
+		});
+	}
+
+	function calculateStatsAverage(graphData) {
+		for (let stat in graphData.data) {
+			const statData = graphData.data[stat];
```
Comment (Sashank999) [inline extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:36]: You could use a `for (const ... of ...)` loop here.

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+				// Calcualte Average
+				let timeLength = (upperTime - lowerTime) / (60 * 60 * 24);
+				let difference = upperVal - lowerVal;
+				let avg = difference / timeLength;
+				let roundedAvg = avg.toFixed(2); // Rounds to 2 decimal places
+				let formattedAvg = roundedAvg.replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+
+				// Insert the data
+				let element = document.querySelectorAll("div[class^='titleItem']")[userIndex];
```
Comment (Sashank999) [inline extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:56]: Variables that do not change will always be declared with `const`, not `let`. Please be consistent with the code base.

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+				element.textContent = `${userName} ( ${formattedAvg} per day )`;
```
Comment (Sashank999) [inline extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:57]: Remove the spaces after `(` and before `)`.

File: extension/scripts/global/globalData.js [hunk 6]
```diff
@@ -403,6 +403,7 @@ const DEFAULT_STORAGE = {
 				stackingMode: new DefaultSetting({ type: "boolean", defaultValue: false }),
 			},
 			profile: {
+				avgpersonalstats: new DefaultSetting({type: "boolean", defaultValue: false}),
```
Comment (Sashank999) [inline extension/scripts/global/globalData.js:406]: Why are we making it disabled by default ? It uses data on the page and does not require any user action. We can make it enabled by default.

**Review by Sashank999 — CHANGES_REQUESTED** [review]
Sorry about the nitpicking. I try to keep code consistent with the code base.

After these changes, I think this feature is good to merge. Thank you for contributing.

File: extension/scripts/global/globalData.js (hunk 6 above)
Comment (DeKleineKobini) [reply extension/scripts/global/globalData.js:406]: I don't agree with this, this is a fairly niche feature. The majority of our users don't need it, and won't understand what that number is doing there. I would keep it disabled by default.

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js (hunk 3 above)
Comment (ibixina) [reply extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:36]: can you clarify what you mean here?

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js (hunk 4 above)
Comment (ibixina) [reply extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:56]: will keep this in mind.

File: extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js (hunk 3 above)
Comment (Sashank999) [reply extension/scripts/features/avg-personal-stat/avg-personal-stat.entry.js:36]: As with the `const` rule for variables that will not be modified later, `for` loop variables must also be declared `const`.

File: extension/scripts/global/globalData.js (hunk 6 above)
Comment (Sashank999) [reply extension/scripts/global/globalData.js]: I agree with DeKleineKobini too now. We can keep this disabled by default.

**Review by Sashank999 — APPROVED** [review]

**Pull request merged**